doc/arkode/guide/source/Usage/LSRKStep/User_callable.rst

```diff
@@ -0,0 +1,394 @@
+
+   .. note:: If one of these set method routines is not called, then LSRKStep will use the 
+      :c:enumerator:`ARKODE_LSRK_RKC_2` by default.
+
+
+.. c:function:: int LSRKStepSetSSPMethodByName(void* arkode_mem, const char* emethod);
+
+   This function selects the LSRK SSP method by name.  The list of allowable values for this input is above.
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the LSRKStep memory block.
+      * *emethod* -- Type of the method in strings.
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARK_ILL_INPUT* if an argument had an illegal value (e.g. typo in the method type).
+
+   .. note:: If one of these set method routines is not called, then LSRKStep will use the 
+      :c:enumerator:`ARKODE_LSRK_SSP_S_2` by default.
+
+
+.. c:function:: int LSRKStepSetDomEigFn(void* arkode_mem, ARKDomEigFn dom_eig);
+
+   Specifies the dominant eigenvalue approximation routine to
+   be used for determining the number of stages that will be used by either the
+   RKC or RKL methods.
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the LSRKStep memory block.
+      * *dom_eig* -- name of user-supplied dominant eigenvalue approximation function (of type :c:func:`ARKDomEigFn()`).
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARKLS_MEM_NULL* if ``arkode_mem`` was ``NULL``.
+      * *ARK_ILL_INPUT* ``dom_eig = NULL`` and LSRKStep does not currently estimate this internally.
+
+   .. note:: This function is currently required when either the RKC or RKL methods are used.
+
+
+.. c:function:: int LSRKStepSetDomEigFrequency(void* arkode_mem, int nsteps);
+
+   Specifies the number of steps after which the dominant eigenvalue information is
+   considered out-of-date, and should be recomputed. This only applies to RKL and RKC methods.
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the LSRKStep memory block.
+      * *nsteps* -- the dominant eigenvalue re-computation update frequency.  A value  ``nsteps = 0``  indicates that the dominant eigenvalue will not change throughout the simulation.
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARKLS_MEM_NULL* if ``arkode_mem`` was ``NULL``.
+
+.. note:: If LSRKStepSetDomEigFrequency routine is not called, then the default ``nsteps`` is set to :math:`25` as recommended in :cite:p:`VSH:04`. 
+   Calling this function with ``nsteps < 0`` resets the default value while ``nsteps = 0`` refers to constant dominant eigenvalue.
+
+
+.. c:function:: int LSRKStepSetMaxNumStages(void* arkode_mem, int stage_max_limit);
+
+   Specifies the maximum number of stages allowed within each time step.  This bound only applies to
+   RKL and RKC methods.
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the LSRKStep memory block.
+      * *stage_max_limit* -- maximum allowed number of stages :math:`(>=2)` and :math:`(<=10000)`.
```

Comment:
We have documented this behavior. I think a limit of `10,000` is still quite high for this context either. Since `λΔt` is proportional to the square of the number of stages `s`, this limit essentially implies `λ ~ -10^8/Δt`. With careful consideration, we may find that round-off error could ultimately limit our results. On the other hand, for the same reasoning I think `200` is a fair default value. This is also the limit mentioned in the paper for efficiency.

Reply:
I agree with @Steven-Roberts that there's no need for us to demand an upper bound on this.  If the behavior is clearly documented, then if a user explicitly requests a larger number of stages we should let them do it.  

Perhaps a middle ground solution would be to include a `.. note::` explaining that the input value translates into a maximum h\lambda \sim -stage_max_limit^2, and that stage_max_limit=200 is the recommended value from the literature (including a citation)?

Reply:
If the user does not specify a value, we should assign a default. I suggest setting `stage_max_limit=200` as the default and allowing users to choose a different value without any limit, or having some large limit like 10^8, which really means `λΔt ~ eps_M`.

Reply:
I mean if the user doesn't call the set function, we still have to assign a number. It is my understanding from your comment that you are fine with having `200` as our default unless otherwise (>=2) is set.